Lib/Protocols/IdHL7.pas

```diff
@@ -919,34 +1062,51 @@ procedure TIdHL7.StopServer;
     end;
 end;
 
-procedure TIdHL7.ServerConnect(AContext: TIdContext);
+Procedure TIdHL7.ServerConnect(AThread: TIdPeerThread);
 var
   LNotify : Boolean;
   LConnCount : integer;
   LValid : Boolean;
+  sIp : String;
+  iIp : Cardinal;
 begin
   assert(Assigned(Self));
-  assert(Assigned(AContext));
+  Assert(Assigned(AThread));
+  //assert(Assigned(AContext));
   assert(Assigned(FLock));
+  LConnCount := 0;
+  sIp := (AThread.Connection.IOHandler as TIdIOHandlerSocket).Binding.PeerIP;

```

Comment:
Rather than type-casting the `IOHandler`, use the public `TIdTCPConnection.Socket` property instead.  Feel free to validate that `Socket` is not `nil`, since a non-socket `IOHandler` could be assigned for debugging purposes.

Reply:
Changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9